From 629f15259487cd602f09fc27464bc013f5a07906 Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@cloudera.com>
Date: Wed, 14 Mar 2012 03:04:43 +0530
Subject: [PATCH 061/795] MR1: MAPREDUCE-4001. Improve MAPREDUCE-3789's fix logic by looking at job's slot demands instead.

Reason: Customer request
Author: Harsh J
Ref: CDH-4276
(cherry picked from commit 99a503c96beae66fa4a153a505dd9f2dae76ad64)
(cherry picked from commit 893b4810fa969aee0ade8ffae3b370528f5a1e77)
---
 .../hadoop/mapred/CapacityTaskScheduler.java       |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/contrib/capacity-scheduler/src/java/org/apache/hadoop/mapred/CapacityTaskScheduler.java b/src/contrib/capacity-scheduler/src/java/org/apache/hadoop/mapred/CapacityTaskScheduler.java
index 2285fb2..f7d60bf 100644
--- a/src/contrib/capacity-scheduler/src/java/org/apache/hadoop/mapred/CapacityTaskScheduler.java
+++ b/src/contrib/capacity-scheduler/src/java/org/apache/hadoop/mapred/CapacityTaskScheduler.java
@@ -344,7 +344,7 @@ private TaskLookupResult getTaskFromQueue(TaskTracker taskTracker,
           // starved
           if ((getPendingTasks(j) != 0 &&
               !hasSufficientReservedTaskTrackers(j)) &&
-                (taskTracker.getAvailableSlots(type) !=
+                !(j.getNumSlotsPerTask(type) >
                  getTTMaxSlotsForType(taskTrackerStatus, type))) {
             // Reserve all available slots on this tasktracker
             LOG.info(j.getJobID() + ": Reserving "
-- 
1.7.0.4

